### PR TITLE
V1 invoicing search invoices

### PR DESCRIPTION
**Pull request by Jdww**
Documented all endpoints to search invoices

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -1159,9 +1454,9 @@ Delete a file
 + date_formatted: "04/04/2017" (string)
 + date_paid: `-1` (number)
 + paid: 0 (number)
-+ total_price_excl_vat: "123.00" (string)
-+ total_price_incl_vat: "130.38" (string)
-+ name: "De Rode Duivles" (string)
++ total_price_excl_vat: 123.00 (number)
++ total_price_incl_vat: 130.38 (number)
++ name: "De Rode Duivels" (string)
```
Comment (Steerlin) [inline apiary.apib]: maybe avoid dutch (examples) in API docs and aim to be international ?

File: src/datastructures.apib [hunk 2]
```diff
@@ -80,16 +80,36 @@
 + due_date_formatted: "25/04/2017" (string)
 + date_paid: `-1` (number)
 + paid: 0 (number)
-+ total_price_excl_vat: "123.00" (string)
-+ total_price_incl_vat: "130.38" (string)
-+ name: "De Rode Duivles" (string)
++ total_price_excl_vat: 123.00 (number)
++ total_price_incl_vat: 130.38 (number)
++ name: "De Rode Duivels" (string)
```
Comment (Steerlin) [inline src/datastructures.apib]: avoid dutch

File: src/datastructures.apib [hunk 3]
```diff
@@ -98,9 +118,9 @@
 + date_formatted: "04/04/2017" (string)
 + date_paid: `-1` (number)
 + paid: 0 (number)
-+ total_price_excl_vat: "123.00" (string)
-+ total_price_incl_vat: "130.38" (string)
-+ name: "De Rode Duivles" (string)
++ total_price_excl_vat: 123.00 (number)
++ total_price_incl_vat: 130.38 (number)
++ name: "De Rode Duivels" (string)
```
Comment (Steerlin) [inline src/datastructures.apib]: same remark

File: src/invoices.apib [hunk 4]
```diff
@@ -250,3 +250,278 @@ Search creditnotes
...
+        + invoice_nr_detailed: `2017 / 5 (MAIN)` (string)
+        + booked: 0 (enum[number])
+            + Members
+                + 0
+                + 1
+        + date: 1491343199 (number)
+        + date_formatted: "04/04/2017" (string)
+        + due_date: 1493157599 (number)
+        + due_date_formatted: "25/04/2017" (string)
+        + date_paid: `-1` (number)
```
Comment (Steerlin) [inline src/invoices.apib]: number so loose back-ticks ? 😕

File: src/invoices.apib [hunk 5]
```diff
@@ -250,3 +250,278 @@ Search creditnotes
...
+        + discount_info (object)
+            + discounts_excl_vat_by_vat_rate (array[number])
+            + discounts_on_vat_by_vat_rate (array[number])
+        + comments: "Comments on the invoice" (string)
+        + invoice_on_hold: 0 (enum[number])
+            + Members
+                + 0
+                + 1
+        + client_name: "De Rode Duivels" (string)
+        + for_attention_of: "Radja Nainggolan" (string)
```
Comment (Steerlin) [inline src/invoices.apib]: not very international example, plus other EU countries might hate you for this when he scores ... 
propose: 'Joe Black'

File: src/invoices.apib (hunk 4 above)
Comment (Jdww) [inline src/invoices.apib]: It's not 100% clear to me when to use back-ticks, a lot of numbers in this and other files have back-ticks

Comment (fowbi) [inline src/invoices.apib]: According to the MSON spec, backticks are only mandatory when the name or literal use reserved keywords or special characters.

> https://github.com/apiaryio/mson/blob/master/MSON%20Specification.md#6-reserved-characters--keywords

In this case it should contain backticks as `-` is used.

File: src/invoices.apib (hunk 5 above)
Comment (svenlombaert) [inline src/invoices.apib]: I tend to use everything related to Pied Piper (Silicon Valley) 😉

**Pull request merged**